src/audio/dai-zephyr.c

```diff
@@ -553,7 +553,7 @@ static int dai_verify_params(struct dai_data *dd, struct comp_dev *dev,
 		return -EINVAL;
 	}
 
-	if (hw_params.channels && hw_params.channels != params->channels) {
+	if (hw_params.channels && !(hw_params.channels >= params->channels)) {
```

Comment:
1. if this is the logic, why not:
    (hw_params.channels && (hw_params.channels < params->channels))

2. please also update the err log message, it is still use "match" words.

Reply:
Thanks for feedback.

1. I agree, it does look better with your version, thank you.

2. In my opinion, the log is appropriate. It specifies that the pcm channels value is incorrect. We could, of course, add a note that the pcm channels value should be less than or equal to the hardware channels value, but this would make the log much longer. At this point, both values are printed, so they should be useful during debugging.
Of course, if you have a suggestion for presenting this log differently, I am open to it :)